### ISSUE TITLE
Add ability to stash individual files

### ISSUE THREAD

**Issue by warmwaffles** [sublimehq/sublime_merge#2]
**Problem description**

When I am working on a feature branch, sometimes I get a little carried away and begin working on something that needs to be isolated from the rest of the pack, but only realize it too late. Right now stashing just stashes the entire set of unstaged or indexed files.

**Preferred solution**

An ability to selectively stash hunks or whole files to be applied to another branch later.

**Alternatives**

Ability to break out individual files into a new branch.

**Comment by deathaxe** [comment]
... and individual lines, so that stashing works with the same amount features as staging.

**Comment by warmwaffles** [comment]
> and individual lines, so that stashing works with the same amount features as staging.

That is true, I have been having to add a temporary commit with the staged lines. stash all other changes, reset back one step, stash it, go to new branch, pop stash and commit, then go back to the previous branch and pop my stash back onto the branch.

**Comment by nikitavoloboev** [comment]
Would love this as well. Ideally with a hotkey similar to `shift+return` that will instead of staging an active diff, would stash it. Perhaps `cmd+shift+return`.

**Comment by willrowe** [comment]
Being able to stash individual files/hunks/lines would be extremely useful, especially if you could incrementally go through add to the "same" stash without needing to go through the stash options for each file/hunk/line, similar to how `git stash -p` works.

**Comment by warmwaffles** [comment]
Today I learned that `git stash -p` exists.

**Comment by michaelblyons** [comment]
Honorable mention: `git stash push -- some_pathspec` (Note: This _does not work_ with `git stash save`.)
